switch_test.go

```diff
@@ -205,8 +214,14 @@ func TestConnPubKeyFilter(t *testing.T) {
 	})
 
 	// connect to good peer
-	go s1.AddPeerWithConnection(c1, false) // AddPeer is blocking, requires handshake.
-	go s2.AddPeerWithConnection(c2, true)
+	go func() {
+		peer, _ := newPeerFromExistingConn(c1, false, s1.reactorsByCh, s1.chDescs, s1.StopPeerForError, s1.config, s1.nodePrivKey)
```

Comment:
we should have AddPeerWithConnection as a helper function that just calls newPeerFromExistingConn and AddPeer. Then we don't have to get so messy here :)